### PR TITLE
Cleanup / add firebase to project

### PR DESCRIPTION
**Pull request by indydcarr**
I added poetry for package management and added firebase / some other tools we needed

@lmason2 if you wanna add the packages that are using in your other code that I missed before we merge this in you can just run 
`poetry install` and then `poetry add <package_name>`

### CODE REVIEW

File: .flake8
Comment (indydcarr) [inline .flake8:1]: Using flake for styling

File: Backyard/Auth/authn.py
Comment (indydcarr) [inline Backyard/Auth/authn.py:1]: we'll use this file for validating firebase login tokens sent from the front end

File: Backyard/conf.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    test_auth_success: bool = False
+    test_mode: bool = False
+    test_session: Any = None
+
+    # -- redis cache --
+
+    redis_enabled: bool = os.getenv("REDIS_ENABLED", True)
+    redis_host: str = os.getenv("REDIS_HOST")
+    redis_port: str = os.getenv("REDIS_PORT")
+    redis_expiration: str = os.getenv("REDIS_EXPIRATION")
```
Comment (indydcarr) [inline Backyard/conf.py:38]: keeping this stuff for now in case we decide to implement caching

File: Backyard/generate_schema.py
Comment (indydcarr) [inline Backyard/generate_schema.py:1]: I'm guessing we'll want an openapi generated eventually for documentation

**Review by lmason2 — APPROVED** [review]

**Pull request merged**